### PR TITLE
perf(table): AVX-512BW 64-byte lane for longest_shared_prefix_length

### LINKED ISSUE #379: perf(table): AVX-512BW 64-byte lane for longest_shared_prefix_length
## Problem

`longest_shared_prefix_length` (src/table/util.rs) is on the block-encoding hot path (flush + compaction): every prefix-truncated entry pays one call against the restart base key. The current x86_64 dispatch tops out at AVX2 (32-byte lanes). On workloads where adjacent keys share long prefixes (time-series keys, namespaced/tenant-prefixed keys, sorted UUIDs with a common high prefix) the shared run frequently exceeds 32 bytes, so a wider lane reduces the number of vector iterations.

## Solution

Add an AVX-512BW kernel (`lsp_avx512`) processing 64 bytes per iteration via `_mm512_cmpeq_epi8_mask` (returns a `__mmask64`; first mismatch = `(!mask).trailing_zeros()`). Extend the runtime dispatch chain to:

```
x86_64: avx512bw (64B) -> avx2 (32B) -> sse2 (16B)
```

- Runtime detection only: `is_x86_feature_detected!("avx512bw")` (cached CPUID), never compile-time `#[cfg(target_feature)]` — the same binary must run on CPUs with and without AVX-512.
- AVX-512BW implies AVX-512F; `_mm512_cmpeq_epi8_mask` is the BW intrinsic. MSRV 1.92 satisfies the 1.89 AVX-512 stabilization.
- Short keys (shared prefix < 64B) fall through to the existing byte-stride tail: zero regression.
- Majority AVX2-only hosts (consumer Intel 11th gen+ dropped AVX-512; AMD Zen4+ and Intel server have it) pay one extra cached atomic load on dispatch, then take the AVX2 path unchanged.

## Acceptance criteria

- [ ] `lsp_avx512` kernel, `#[target_feature(enable = "avx512bw")]`, byte-stride tail
- [ ] dispatch checks avx512bw first, falls back to avx2 then sse2
- [ ] direct kernel test gated on `is_x86_feature_detected!("avx512bw")` (mirrors `lsp_avx2_matches_reference_on_boundaries`), reusing `assert_kernel_matches_reference`
- [ ] proptest + boundary tests already cover the dispatched path; verify they pass on an AVX-512 host
- [ ] clippy --all-features --all-targets clean, fmt clean, rustdoc clean

## Estimate

3h (kernel + dispatch + direct test + doc update; reuses existing test harness)

### PR DESCRIPTION
**Pull request by polaz**
## Summary

Adds an AVX-512BW kernel to `longest_shared_prefix_length` (the block-encoding hot path used during flush + compaction), extending the x86_64 runtime dispatch from AVX2 to a wider 64-byte lane.

```
x86_64 dispatch: avx512bw (64B) -> avx2 (32B) -> sse2 (16B)
```

`lsp_avx512` processes 64 bytes per iteration via `_mm512_cmpeq_epi8_mask`, which folds the lane comparison straight into a native `__mmask64` (no separate `movemask` step). First mismatch is `(!mask).trailing_zeros()`; full match is `u64::MAX`. The structure mirrors the proven AVX2 kernel.

## Why

Workloads where adjacent keys share long prefixes (time-series keys, tenant/namespace-prefixed keys, sorted UUIDs with a common high prefix) frequently have shared runs exceeding 32 bytes. A 64-byte lane halves the vector-iteration count on those runs. Keys with shared prefix < 64B fall through to the existing byte-stride tail, so there is no regression for short keys.

## Detection

- Runtime-only: `is_x86_feature_detected!("avx512bw")` (cached CPUID). No compile-time `#[cfg(target_feature)]` — the same binary runs on CPUs with and without AVX-512.
- AVX-512BW implies the F subset; `_mm512_cmpeq_epi8_mask` is the BW byte-granular compare. MSRV 1.92 satisfies the 1.89 AVX-512 stabilization.
- Majority AVX2-only hosts (consumer Intel 11th gen+ dropped AVX-512; AMD Zen4+ / Intel server keep it) pay one extra cached atomic load on dispatch, then take the AVX2 path unchanged.

## Testing

- `cargo nextest run --all-features`: 1801 passed, 2 skipped
- `cargo clippy --all-features --all-targets -- -D warnings` clean on both `aarch64-apple-darwin` and `x86_64-apple-darwin`
- `cargo fmt --check` clean, `RUSTDOCFLAGS="-D warnings" cargo doc --no-deps --all-features` clean
- New direct kernel test `lsp_avx512_matches_reference_on_boundaries` gated on `is_x86_feature_detected!("avx512bw")`, reusing `assert_kernel_matches_reference`; runs and verifies the kernel on any AVX-512BW CI runner, no-op otherwise. Existing proptest + boundary tests cover the dispatched path.

Note: the AVX-512 execution path cannot be exercised on the author's aarch64 host (Rosetta lacks AVX-512); CI x86_64 runners with AVX-512BW provide live coverage.

Closes #379

## Summary by CodeRabbit

* **Refactor**
  * Optimized string comparison operations for x86_64 systems with newer CPU capabilities, improving performance through enhanced instruction dispatch that automatically selects the best code path based on processor features.

* **Tests**
  * Extended test coverage to validate optimization across boundary conditions and mismatch scenarios on compatible hardware.

* **Documentation**
  * Updated dispatch strategy documentation and feature-detection caching behavior.

### CODE REVIEW

File: src/table/util.rs [hunk 1]
```diff
@@ -793,6 +860,22 @@ mod tests {
...
     }
 
+    /// Direct test for the AVX-512BW 64-byte kernel, gated by runtime CPU detection.
+    /// CI `x86_64` runners may or may not expose AVX-512 (consumer Intel 11th gen+
+    /// dropped it; AMD Zen4+ and Intel server keep it), so the dispatched path can't
+    /// be relied on to reach `lsp_avx512`. This direct test exercises it whenever the
+    /// host supports AVX-512BW and is a no-op otherwise.
+    #[cfg(target_arch = "x86_64")]
+    #[test]
+    fn lsp_avx512_matches_reference_on_boundaries() {
```
[the remark this answers is not in the document]
Comment (polaz) [reply src/table/util.rs]: Renamed to `lsp_avx512_on_boundaries_matches_reference`. Also normalized the sibling boundary tests (scalar/sse2/avx2/neon + the dispatch test) to the same `<what>_<condition>_<expected>` order so the whole group stays consistent rather than leaving one outlier.
